src/portable_python/__init__.py

```diff
@@ -680,6 +680,16 @@ def compile(self):
                 runez.decompress(path, self.m_src_build, simplify=True)
 
                 env_vars = self._get_env_vars()
+                if not PPG.config.get_value("allow-homebrew"):
+                    # Remove any mention of /opt/homebrew from PATH (reduce chances of dynamic links to homebrew)
+                    path_env_var = env_vars.get("PATH")
+                    if path_env_var:
```

Comment:
This is gated by the presence of configured env_vars but mutates the `os.environment` vars. Shouldn't this:

a. be done regardless of configured env vars
b. take into account the configured paths instead of wholesale overriding them

Reply:
The mechanism used takes into account existing env vars, configured env vars, and programmatically yielded env vars. The idea was to do similarly to how most build scripts are written, typically like `"CFLAGS=${CFLAGS} -Lmy-flag"`.

One can always do this to just set an env var to something (without following this "additive" behavior that p-p has):
```
export PATH=...
portable-python ...
```

But you're right I could do this at startup actually 🤔

b) is already the case (p-p follows the same "additive" approach for all env vars)

a) -> yup, I could indeed "groom" PATH at startup instead of here

Reply:
Actually I think it's best if we do it like this here, it does allow for an "advanced" case like for example this config:
```
macos:
  env:
    MACOSX_DEPLOYMENT_TARGET: 13  # Ventura
  openssl:
    allow-homebrew: true
```

Now, homebrew is allowed only on macos AND openssl only (every other component will have `/opt/homebrew` cleaned up from PATH by default)